### PR TITLE
Adds CAPNP_BEGIN_HEADER and CAPNP_END_HEADER around schema headers

### PR DESCRIPTION
**Pull request by ThatGeoGuy**
In reference to an email I sent to the mailing list: https://groups.google.com/d/msg/capnproto/NN5LXfR5ngw/Jf8ACrWzBAAJ

This should bound any instance of namespaces for `capnp::schemas` with `CAPNP_BEGIN_HEADER` and `CAPNP_END_HEADER` like so:

```
CAPNP_BEGIN_HEADER

namespace capnp {
namespace schemas {

} // namespace schemas
} // namespace capnp

CAPNP_END_HEADER

```

### CODE REVIEW

File: c++/src/capnp/compiler/capnpc-c++.c++ [hunk 1]
```diff
@@ -3045,13 +3045,17 @@ private:
...
+          "\n"
           "namespace capnp {\n"
           "namespace schemas {\n"
           "\n",
           KJ_MAP(n, nodeTexts) { return kj::mv(n.capnpSchemaDecls); },
           "\n"
           "}  // namespace schemas\n"
           "}  // namespace capnp\n"
+          "\n"
+          "CAPNP_END_HEADER\n"
```
Comment (kentonv) [inline c++/src/capnp/compiler/capnpc-c++.c++:3058]: Looks like you've placed this at the end of the schema block, but the main body of the header actually appears after that. The last part of the header is generated at line 3067 below, so you'll want to insert this after that.

**Comment by kentonv** [comment]
You'll need to run `regenerate-bootstraps.sh` to regenerate all the generated code (like `schema.capnp.h`) using the modified compiler.

File: c++/src/capnp/compiler/capnpc-c++.c++ (hunk 1 above)
Comment (ThatGeoGuy) [reply c++/src/capnp/compiler/capnpc-c++.c++]: Should be resolved as of 4df555139a3feaaf9746740cefecf0beadc01ba9

**Comment by ThatGeoGuy** [comment]
Resolved the above feedback and re-ran `regenerate-bootstraps.sh`. 

Do let me know what else may be needed!

**Comment by kentonv** [comment]
Ugh, looks like our MSVC and Cygwin tests have both bitrotted -- not your fault.

Looks good, thanks for doing this!

**Pull request merged**